File: src/main/java/rs/helpkit/util/fx/GraphicsState.java
package rs.helpkit.util.fx;

import java.awt.*;
import java.awt.geom.AffineTransform;

/**
 * @author Tyler Sedlar
 * @since 1/20/2016
 */
public class GraphicsState {

    private final Graphics2D g;
    private final Color color;
    private final Color background;
    private final Shape clip;
    private final Composite composite;
    private final Font font;
    private final Paint paint;
    private final RenderingHints hints;
    private final Stroke stroke;
    private final AffineTransform transform;

    public GraphicsState(Graphics2D g) {
        this.g = g;
        this.color = g.getColor();
        this.background = g.getBackground();
        this.clip = g.getClip();
        this.composite = g.getComposite();
        this.font = g.getFont();
        this.paint = g.getPaint();
        this.hints = g.getRenderingHints();
        this.stroke = g.getStroke();
        this.transform = g.getTransform();
    }

    public void restore() {
        g.setColor(color);
        g.setBackground(background);
        g.setClip(clip);
        g.setComposite(composite);
        g.setFont(font);
        g.setPaint(paint);
        g.setRenderingHints(hints);
        g.setStroke(stroke);
        g.setTransform(transform);
    }
}

File: src/main/java/rs/helpkit/util/fx/GraphicsState.kt
package rs.helpkit.util.fx

import java.awt.*
import java.awt.geom.AffineTransform

/**
 * @author Tyler Sedlar
 * @since 1/20/2016
 */
class GraphicsState(private val g: Graphics2D) {
    private val color: Color? = g.color
    private val background: Color? = g.background
    private val clip: Shape? = g.clip
    private val composite: Composite? = g.composite
    private val font: Font? = g.font
    private val paint: Paint? = g.paint
    private val hints: RenderingHints? = g.renderingHints
    private val stroke: Stroke? = g.stroke
    private val transform: AffineTransform? = g.transform

    fun restore() {
        g.color = color
        g.background = background
        g.clip = clip
        g.composite = composite
        g.font = font
        g.paint = paint
        g.setRenderingHints(hints)
        g.stroke = stroke
        g.transform = transform
    }
}
